gax/src/test/java/com/google/api/gax/rpc/OperationCallableImplTest.java

```diff
@@ -0,0 +1,983 @@
+        OperationSnapshot response = results[index];
+        if (index < results.length - 1) {
+          index += 1;
+        }
+        return newFuture(returnStatusCode, response);
+      }
+    };
+  }
+
+  private UnaryCallable<Integer, OperationSnapshot> getUnexpectedStartCallable() {
+    return new UnaryCallable<Integer, OperationSnapshot>() {
+      @Override
+      public ApiFuture<OperationSnapshot> futureCall(Integer request, ApiCallContext context) {
+        return ApiFutures.immediateFailedFuture(
+            new UnsupportedOperationException("Unexpected call to start operation"));
+      }
+    };
+  }
+
+  private class UnsupportedOperationApi implements OperationApi {
+    @Override
+    public UnaryCallable<String, OperationSnapshot> getOperationCallable() {
+      throw new UnsupportedOperationException("Didn't expect call to getOperationCallable()");
+    }
+
+    @Override
+    public UnaryCallable<String, Void> cancelOperationCallable() {
+      throw new UnsupportedOperationException("Didn't expect call to cancelOperationCallable()");
+    }
+
+    @Override
+    public UnaryCallable<String, Void> deleteOperationCallable() {
+      throw new UnsupportedOperationException("Didn't expect call to deleteOperationCallable()");
+    }
+  }
+
+  private OperationApi mockGetOperation(
+      final Code returnStatusCode, final OperationSnapshot... results) {
+    return new UnsupportedOperationApi() {
+      private UnaryCallable<String, OperationSnapshot> getOperationCallable =
+          mockGetOpSnapshotCallable(returnStatusCode, results);
+
+      @Override
+      public UnaryCallable<String, OperationSnapshot> getOperationCallable() {
+        return getOperationCallable;
+      }
+    };
+  }
+
+  private OperationApi mockCancelOperation(final Code returnStatusCode) {
+    return new UnsupportedOperationApi() {
+      private UnaryCallable<String, Void> cancelOperationCallable =
+          new UnaryCallable<String, Void>() {
+            @Override
+            public ApiFuture<Void> futureCall(String request, ApiCallContext context) {
+              return newFuture(returnStatusCode, null);
+            }
+          };
+
+      @Override
+      public UnaryCallable<String, Void> cancelOperationCallable() {
+        return cancelOperationCallable;
+      }
+    };
+  }
+
+  private <ResponseT> ApiFuture<ResponseT> newFuture(Code returnStatusCode, ResponseT response) {
+    if (Code.OK.equals(returnStatusCode)) {
+      return ApiFutures.immediateFuture(response);
+    } else {
+      return ApiFutures.immediateFailedFuture(
+          FakeApiExceptionFactory.createException(null, returnStatusCode, false));
+    }
+  }
+
+  private void assertExceptionMatchesCode(FakeStatusCode code, Throwable exception) {
+    Class expectedClass;
+    switch (code.getCode()) {
+      case CANCELLED:
+        expectedClass = CancelledException.class;
+        break;
+        //      case NOT_FOUND:
```

Comment:
Why are all these commented?

Reply:
I copied the function from somewhere else, and commented out the enum values that I hadn't defined yet in `FakeStatusCode.Code`.